### PR TITLE
DM: Fixes issue #592 NHLT is not detected on audio passthru.

### PR DESCRIPTION
**Pull request by javedm**
When XSDT is created by DM, if audio passthru is enabled for
audio device (0:e:0), an entry is added in XSDT that references
the NHLT table. With this fix, NHLT appears in the kernel boot
log for ACPI and entry can be seen in /sys/firmware/acpi/tables/NHLT.

Signed-off-by: Madeeha Javed <madeeha_javed@mentor.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

**Comment by dongyaozu** [comment]
@gzhai7 can you review?

File: devicemodel/hw/platform/acpi/acpi.c [hunk 1]
```diff
@@ -198,6 +200,10 @@ basl_fwrite_xsdt(FILE *fp, struct vmctx *ctx)
 	EFPRINTF(fp, "[0004]\t\tACPI Table Address 3 : 00000000%08X\n",
 	    basl_acpi_base + MCFG_OFFSET);
 
+	if (acpi_is_table_valid(NHLT_ENTRY_NO))
```
[the remark this answers is not in the document]
Comment (javedm) [reply devicemodel/hw/platform/acpi/acpi.c]: done

**Comment by javedm** [comment]
@gzhai7 yes we are using the old audio driver with AUDIO_NHLT_HACK. Your suggested changes are done. Thanks a lot for reviewing the patch.

**Comment by gzhai7** [comment]
Sorry, seems my comments is not visual to you. Could you pls. change the function name?
acpi_is_table_valid => acpi_table_is_valid
Thanks,

File: devicemodel/hw/platform/acpi/acpi.c [hunk 2]
```diff
@@ -976,6 +982,12 @@ acpi_table_enable(int num)
 	basl_ftables[num].valid = true;
 }
 
+static bool
+acpi_is_table_valid(int num)
```
[the remark this answers is not in the document]
Comment (dongyaozu) [reply devicemodel/hw/platform/acpi/acpi.c]: should be  "is_acpi_table_valid" ?

**Comment by javedm** [comment]
Sorry for missing this change earlier. I have now changed it to acpi_table_is_valid. Thanks!

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**